src/sql/base/browser/ui/table/table.ts

```diff
@@ -130,8 +136,22 @@ export class Table<T extends Slick.SlickData> extends Widget implements IDisposa
 		this.mapMouseEvent(this._grid.onDblClick, this._onDoubleClick);
 		this._grid.onColumnsResized.subscribe(() => this._onColumnResize.fire());
 
-		this._grid.onKeyDown.subscribe((e, args: Slick.OnKeyDownEventArgs<T>) => {
+		this._grid.onKeyDown.subscribe((e: Slick.EventData, args: Slick.OnKeyDownEventArgs<T>) => {
 			const evt = (e as JQuery.Event).originalEvent as KeyboardEvent;
+			if (evt.key === 'F3' && options.useF3KeyToSortColumns) {
```

Comment:
you should not introduce a new option, if sorting is enabled then the keyboard shortcut should just work.
1. Add the handler to the place where the sorting by mouse click is handled and reuse the code 
2. Don't hardcode the shortcut key - I can take care of this in a future PR, I have similar work planned for this milestone.